src/libmongoc/src/mongoc/mongoc-async-cmd-private.h

```diff
@@ -265,8 +267,13 @@ _acmd_has_timed_out (const mongoc_async_cmd_t *self)
 static inline void
 _acmd_cancel (mongoc_async_cmd_t *self)
 {
-   // XXX: Should this check if ther command has already finished/failed?
-   self->state = MONGOC_ASYNC_CMD_CANCELLED_STATE;
+   BSON_ASSERT_PARAM (self);
+
+   // Don't attempt to cancel a comman in the error state, as it will already have
+   // a waiting completion.
```

Comment:
```suggestion
   // Don't cancel a command in the error state, as it already has a waiting completion.
```
Wording tweak suggestion + perhaps replace "waiting completion" (new term not found elsewhere) with a different description?